modules/junit-vintage/src/main/java/org/testcontainers/junit/vintage/TemporaryNetwork.java

```diff
@@ -0,0 +1,61 @@
+package org.testcontainers.junit.vintage;
+
+import org.junit.rules.ExternalResource;
+import org.testcontainers.containers.Network;
+
+/**
+ * Integrates {@link Network} with the JUnit4 lifecycle.
+ */
+public final class TemporaryNetwork extends ExternalResource implements Network {
```

Comment:
Frankly, I believe `Network` deserves its own annotation like `@Container`.
Of course, for `JUnit5` they could use `@AutoClose`, however, it is not future-proof.

As you might have seen, in my "migrate to junit5" I added `@ManagedNetwork` for exactly this reason: make sure the users code against testcontainers-apis rather than junit-apis.

The discussion is here: https://github.com/testcontainers/testcontainers-java/issues/9552#issuecomment-2934998013

Reply:
Interesting idea. I think what I have here would be a very simple migration for users (swapping one class for another). Since this implements `Network` users can still code against Testcontainers APIs.

Leaving this open to see what others think.

Reply:
> However, if the network would need some `before...` handling, then `@AutoClose` alone won't help.

Networks currently do not have before handling. If they do in the future, `TemporaryNetwork` could be updated accordingly. Users concerned about the ordering between the two rules introduced here can use `RuleChain` or the `order` attribute of `@Rule`.

The goal of this PR is to provide APIs for Testcontainers users that are on JUnit4 to migrate to before the core Testcontainers code is updated to remove references to JUnit4 classes. I'm not trying to put together a perfect API, and I certainly do not want this PR to include changes to the JUnit Jupiter integration.

Reply:
Because I want to make a minimal change to allow people to migrate.

If you don't look like this proposal, send another.